integrations/va-okta/src/index.tsx

```diff
@@ -0,0 +1,284 @@
+                    if ('access_token' in resp) {
+                        let url;
+                        const state = request.query.state.toString();
+                        const location = state.substring(state.indexOf('-') + 1);
+                        if (location) {
+                            url = `${spaceData.urls?.published}${location}/?jwt_token=${token}`;
+                        } else {
+                            url = `${spaceData.urls?.published}/?jwt_token=${token}`;
+                        }
+                        if (token && spaceData.urls?.published) {
+                            return Response.redirect(url);
+                        } else {
+                            return new Response(
+                                "Error: Either JWT token or space's published URL is missing",
+                                {
+                                    status: 500,
+                                }
+                            );
+                        }
+                    } else {
+                        return new Response('Error: No Access Token found in response from Okta', {
+                            status: 401,
+                        });
+                    }
+                } else {
+                    return new Response('Error: Either ClientId or Client Secret is missing', {
+                        status: 400,
+                    });
+                }
+            }
+        });
+
+        let response;
+        try {
+            response = await router.handle(request, context);
+        } catch (error: any) {
+            logger.error('error handling request', error);
+            return new Response(error.message, {
+                status: error.status || 500,
+            });
+        }
+
+        if (!response) {
+            return new Response(`No route matching ${request.method} ${request.url}`, {
+                status: 404,
+            });
+        }
+
+        return response;
+    }
+};
+
+export default createIntegration({
+    fetch: handleFetchEvent,
+    components: [configBlock],
+    fetch_visitor_authentication: async (event, context) => {
+        const { environment } = context;
+        const installationURL = environment.spaceInstallation?.urls?.publicEndpoint;
+        const oktaDomain = environment.spaceInstallation?.configuration.okta_domain;
+        const clientId = environment.spaceInstallation?.configuration.client_id;
+        const location = event.location ? event.location : '';
+
+        const url = new URL(`https://${oktaDomain}/oauth2/default/v1/authorize`);
+        url.searchParams.append('client_id', clientId);
+        url.searchParams.append('response_type', 'code');
+        url.searchParams.append('redirect_uri', `${installationURL}/visitor-auth/response`);
+        url.searchParams.append('response_mode', 'query');
+        url.searchParams.append('scope', 'openid');
+        url.searchParams.append('state', `state-${location}`);
```

Comment:
Why does it have to be prefixed with `state-`

Reply:
Woops, somehow missed this comment. It's because Okta does not allow empty string as `state`, whereas Auth0 and Azure allow empty string as `state`.